Magic/src/main/java/com/elmakers/mine/bukkit/action/builtin/ChatAction.java

```diff
@@ -0,0 +1,51 @@
+package com.elmakers.mine.bukkit.action.builtin;
+
+import com.elmakers.mine.bukkit.action.BaseSpellAction;
+import com.elmakers.mine.bukkit.api.action.CastContext;
+import com.elmakers.mine.bukkit.api.spell.SpellResult;
+import org.bukkit.ChatColor;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.entity.Player;
+
+import java.util.List;
+
+public class ChatAction extends BaseSpellAction {
+
+    private String message;
+
+    private String translate(String str) {
+        return ChatColor.translateAlternateColorCodes('&', str);
+    }
+
+    @Override
+    public void prepare(CastContext context, ConfigurationSection parameters) {
+        super.prepare(context, parameters);
+
+        this.message = context.getMessage(parameters.getString("message"), translate(parameters.getString("message")));
+
+        List<String> l = parameters.getStringList("randomized_messages");
+
+        if (!l.isEmpty()) this.message = translate(l.get(context.getRandom().nextInt(l.size())));
```

Comment:
You should also use `#getMessage` here

Reply:
For which? getMessage returns a String and wouldn't support the randomization, would it?

Reply:
It will do lookups in the messages file when you specify an individual message, but not when you specify a list.

Reply:
I'm still a little confused as to what you're suggesting here.

Reply:
I think he means to use Messages to get messages in all cases.

I kind of agree, though I think I set a bad precedent with the Message action.

Messages does actually support randomized messages, too:

https://github.com/elBukkit/MagicPlugin/blob/master/MagicAPI/src/main/java/com/elmakers/mine/bukkit/api/magic/Messages.java#L15

That will return a random string from a list of strings defined in messages.yml

The idea with messages.yml is to keep ALL text in there, if possible, so translations are easier.

I kind of failed on that with the Message action, but that could be cleaned up.

Basically what this would mean for this action is that the parameters are not strings to send in chat, but message keys to look up via context.getMessages().get() or .getRandomized().